lib/web-ledger-client.js

```diff
@@ -0,0 +1,215 @@
+    result.type = 'WebLedgerRecord'; //FIXME: check if type is correct
+    result.record = record;
+
+    return result;
+  }
+
+  /**
+   * Sends an operation to a Web Ledger node.
+   *
+   * @param operation {object} Object to be wrapped in a Web Ledger operation
+   * @param [headers={}] {object} Optional headers to pass through with request
+   *
+   * @returns {Promise<object>} Axios response object
+   */
+  async send({operation, headers = {}}) {
+    const ledgerAgent = await this.getAgent();
+
+    const headers = Object.assign({}, DEFAULT_HEADERS, headers);
+
+    return axios.post(ledgerAgent.service.ledgerOperationService, operation, {
+      httpsAgent: this.httpsAgent,
+      headers
+    });
+  }
+
+  /**
+   * Resolves with a list of ledger agents.
+   *
+   * @returns {Promise<Array<object>>}
+   */
+  async getAgents() {
+    const ledgerAgentsUrl = `https://${this.hostname}/ledger-agents`;
+
+    let ledgerAgentResponse;
+    try {
+      ledgerAgentResponse = await axios.get(ledgerAgentsUrl, {
+        httpsAgent: this.httpsAgent,
+        headers: DEFAULT_HEADERS
+      });
+    } catch(error) {
+      this.logger.error('Could not fetch ledger agents:', error);
+      throw error;
+    }
+
+    return ledgerAgentResponse.data.ledgerAgent;
+  }
+
+  /**
+   * Resolves with a single ledger agent for given options. If "id" is not
+   * specified the first ledger agent will be resolved.
+   *
+   *
+   * @returns {Promise<object>}
+   */
+  async getAgent({id} = {}) {
+    const agents = await this.getAgents();
+
+    if(!id) {
+      return agents[0]; //TODO: Should we be returning the first ledger agent
+    }
+
+    const [agent] = agents.filter(agent => agent.id === id);
+
+    if(!agent) {
+      const error = new Error(`Agent with id "${id}" could not be found`);
+      error.name = 'NotFoundError';
+      this.logger.error('Could not fetch ledger agents:', error);
+      throw error;
+    }
+
+    return agent;
+  }
+
+  static signRequestHeaders({path, headers, signHeaders, keyId, key, method}) {
+    const httpSignature = require('http-signature');
+
+    httpSignature.signRequest({
+      getHeader: (header) => {
+        // case insensitive lookup
+        return headers[Object.keys(headers).find(
+          key => key.toLowerCase() === header.toLowerCase())];
+      },
+      setHeader: (header, value) => {
+        headers[header] = value;
+      },
+      method,
+      path
+    }, {
+      headers: signHeaders,
+      keyId,
+      key
+    });
+  }
+
+  /**
+   * Wraps a Record in a Web Ledger Operation.
+   */
+  static wrap({record, operationType = 'create'}) {
+    const operation = {
+      '@context': WEB_LEDGER_CONTEXT_V1
+    };
+
+    switch(operationType) {
+      case 'create':
+        operation.type = 'CreateWebLedgerRecord';
+        operation.record = record;
+        break;
+      case 'update':
+        operation.type = 'UpdateWebLedgerRecord';
+        operation.recordPatch = record;
```

Comment:
I'm slightly worried about this part, `operation.recordPatch = record`.
Originally it was:

```js
operation.recordPatch = didDocument.commit();
```
And that `commit()` [sets up a change tracker](https://github.com/digitalbazaar/did-io/blob/3e981459f6d7761cc3fe4d9724ba03fa9db1e85d/lib/methods/veres-one/veres-did-doc.js#L193).

@dlongley what's the expectation with updating Web Ledger records? Do they also want a patch tracker?

Reply:
I removed it because I felt that `commit()` was specific to did-io. Will we be able to guarantee that that commit method exists on all records that are passed into this function? More so, would it be sufficient to have web-ledger-client only receive the necessary recordPatch to send to the ledger and have a higher level library initiate the change tracker?

Reply:
I agree with you. I'll do the change tracker thing in client code (in did-client-veres-one).